fury/ui.py

```diff
@@ -4657,6 +4657,361 @@ def left_button_dragged(self, i_ren, _obj, _sub_component):
         i_ren.force_render()
 
 
+class TabUI(UI):
+    """ UI element to add multiple panels within a single window.
+
+    Attributes
+    ----------
+    tabs: :class: List of 'TabPanel2D'
+        Stores all the instances of 'TabPanel2D' that renderes the contents.
+    """
+
+    def __init__(self, position=(0, 0), size=(100, 100), nb_tabs=1,
+                 active_color=(1, 1, 1), inactive_color=(0.5, 0.5, 0.5),
+                 draggable=False):
+        """
+
+        Parameters
+        ----------
+        position : (float, float)
+            Absolute coordinates (x, y) of the lower-left corner of this
+            UI component.
+        size : (int, int)
+            Width and height in pixels of this UI component.
+        nb_tabs : int
+            Number of tabs to be renders.
+        active_color : tuple of 3 floats.
+            Background color of active tab panel.
+        inactive_color : tuple of 3 floats.
+            Background color of inactive tab panels.
+        draggable : {True, False}
+            Whether the UI element is draggable or not.
+        """
+        self.tabs = []
+        self.nb_tabs = nb_tabs
+        self.parent_size = size
+        self.content_size = (size[0], int(0.9 * size[1]))
+        self.draggable = draggable
+        self.active_color = active_color
+        self.inactive_color = inactive_color
+        self.active_tab_idx = None
+        self.collapsed = True
+
+        super(TabUI, self).__init__()
+        self.position = position
+
+    def _setup(self):
+        """ Setup this UI component.
+        Create parent panel.
+        Create tab panels.
+        """
+        self.parent_panel = Panel2D(self.parent_size, opacity=0.0)
+
+        # Offer some standard hooks to the user.
+        self.on_change = lambda ui: None
+        self.on_collapse = lambda ui: None
+
+        for _ in range(self.nb_tabs):
+            content_panel = Panel2D(size=self.content_size)
+            content_panel.set_visibility(False)
+            tab_panel = TabPanel2D(content_panel=content_panel)
+            self.tabs.append(tab_panel)
+        self.update_tabs()
+
+    def _get_actors(self):
+        """ Get the actors composing this UI component.
+        """
+        actors = []
+        actors += self.parent_panel.actors
+        for tab_panel in self.tabs:
+            actors += tab_panel.actors
```

Comment:
let's make this more pythonic with:

`actors = self.parent_panel.actors + [tab_panel.actors for tab_panel in self.tabs]`

if it does not fit in 1 line, you can split it into 2 steps.

Reply:
Hello @skoudoro , this snippet wont work as `tab_panel.actors` returns a list. Therefore after the comprehension is finished a 2D list is generated where each corresponding list contains the actors of the `tab_panel`.
Something like this:
```
[[actors of tab panel 1],
 [actors of tab panel 2],
 ...
[actors of last tab panel]]
```

Reply:
> this snippet wont work as tab_panel.actors returns a list

Oh ok, I did not see it. so no need to bother. you can keep it as it is